### PR TITLE
3.0 migration guide

### PR DESCRIPTION
**Pull request by rnystrom**
Will add to this as we close out the [milestone](https://github.com/Instagram/IGListKit/milestone/3).

Issue fixed: #475

### CODE REVIEW

File: Guides/Migration.md [hunk 1]
```diff
@@ -2,6 +2,61 @@
...
+
+For details on all changes in IGListKit 2.0.0, please see the [release notes](https://github.com/Instagram/IGListKit/releases/tag/3.0.0).
+
+### IGListBindingSectionController
+
+If you were using `IGListDiff(...)` _inside_ a section controller to compute diffs for cells, we recommend that you start using `IGListBindingSectionController` which wraps this behavior in an elegant and tested API.
+
+### IGListCollectionView removed
+
+You can simply Find & Replace `IGListCollectionView` with `UICollectionView` in your project to fix this.
```
Comment (jessesquires) [inline Guides/Migration.md]: this won't work completely. `IGListCollectionView` will match the beginning of IGListCollectionView**Layout**.

need to search `IGListCollectionView *` and `IGListCollectionView alloc]`, etc.

File: Guides/Migration.md [hunk 2]
```diff
@@ -2,6 +2,61 @@
 
 This guide provides details for how to migration between major versions of `IGListKit`.
 
+## From 2.x to 3.x
+
+For details on all changes in IGListKit 2.0.0, please see the [release notes](https://github.com/Instagram/IGListKit/releases/tag/3.0.0).
```
Comment (jessesquires) [inline Guides/Migration.md]: 3.0 😄

Comment (Sherlouk) [inline Guides/Migration.md]: Copy and paste catches the best off guard!

File: Guides/Migration.md [hunk 3]
```diff
@@ -2,6 +2,61 @@
...
+[self.collectionContext performBatchAnimated:YES updates:^(id<IGListBatchContext> batchContext) {
+  self.expanded = YES;
+  [batchContext insertInSectionController:self atIndexes:[NSIndexSet indexSetWithIndex:1]];
+} completion:nil];
+```
+
+**Swift**
+
+```swift
+expanded = true
```
Comment (BasThomas) [inline Guides/Migration.md]: This can use an `// OLD` and `// NEW` for clarity as well.

**Comment by rnystrom** [comment]
Will need `IGListSectionType` stuff once #168 lands:

- Swift needs `override`
- Find/delete anything refing `IGListSectionType`
- (optional) Delete default or empty methods

File: Examples/Examples-iOS/IGListKitExamples/SectionControllers/EmbeddedSectionController.swift [hunk 4]
```diff
@@ -45,6 +45,10 @@ final class EmbeddedSectionController: IGListSectionController, IGListSectionTyp
         number = object as? Int
     }
 
-    func didSelectItem(at index: Int) {}
+    func didSelectItem(at index: Int) {
+        collectionContext?.performBatch(animated: true, updates: { (batchContext) in
+            batchContext.insert(in: self, at: [0])
+        })
+    }
```
Comment (rnystrom) [inline Examples/Examples-iOS/IGListKitExamples/SectionControllers/EmbeddedSectionController.swift]: whoops this is wrong, deleting

**Comment by jessesquires** [comment]
@rnystrom 

We need to mention the Swift names changes (#593), then I think this is probably good to go?

File: Guides/Migration.md [hunk 5]
```diff
@@ -2,6 +2,84 @@
...
 
+## From 2.x to 3.x
+
+For details on all changes in IGListKit 3.0.0, please see the [release notes](https://github.com/Instagram/IGListKit/releases/tag/3.0.0).
+
+### "IG" prefix removed for Swift
+
+Since Swift uses modules, we don't need to use the `IG` prefix to avoid collisions. This also makes your code look _Swifty_.
+
+To migrate, use Xcode's Find navigator (command-3), search for `IGList`, and replace with `List`.
```
Comment (rnystrom) [inline Guides/Migration.md]: @jessesquires think this is enough?

**Comment by jessesquires** [comment]
made some minor edits. going to import and merge.

**Pull request closed without merging**